service/matching/db.go

```diff
@@ -231,6 +234,9 @@ func (db *taskQueueDB) CreateTasks(
 			Tasks: tasks,
 		})
 
+	// Update the maxReadLevel after the writes are completed, but before we send the response,
+	// so that taskReader is guaranteed to see the new read level when SpoolTask wakes it up.
+	db.maxReadLevel = maxReadLevel
```

Comment:
Moved the assignment after `CreatingTasks` just to be consistent with the old code; 

even here, even if the request to `db` fails (for whatever reason), maxReadLevel is updated and that behaviour is kept consistent with ID allocation and maxReadLevel updates of before.

Reply:
thinking more about this comment, it's not relevant anymore because GetMaxReadLevel is locked too. let's get rid of it

Reply:
not that we plan to not lock for GetMaxReadLevel, the order matters, and the comment should be kept.